src/lib.rs

```diff
@@ -1415,16 +1886,258 @@ impl<T> FromIterator<T> for ThinVec<T> {
     }
 }
 
+impl<T: Clone> From<&[T]> for ThinVec<T> {
+    /// Allocate a `ThinVec<T>` and fill it by cloning `s`'s items.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use thin_vec::{ThinVec, thin_vec};
+    ///
+    /// assert_eq!(ThinVec::from(&[1, 2, 3][..]), thin_vec![1, 2, 3]);
+    /// ```
+    fn from(s: &[T]) -> ThinVec<T> {
+        s.iter().cloned().collect()
+    }
+}
+
+#[cfg(not(no_global_oom_handling))]
+impl<T: Clone> From<&mut [T]> for ThinVec<T> {
+    /// Allocate a `ThinVec<T>` and fill it by cloning `s`'s items.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use thin_vec::{ThinVec, thin_vec};
+    ///
+    /// assert_eq!(ThinVec::from(&mut [1, 2, 3][..]), thin_vec![1, 2, 3]);
+    /// ```
+    fn from(s: &mut [T]) -> ThinVec<T> {
+        s.iter().cloned().collect()
+    }
+}
+
+impl<T, const N: usize> From<[T; N]> for ThinVec<T> {
+    /// Allocate a `ThinVec<T>` and move `s`'s items into it.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use thin_vec::{ThinVec, thin_vec};
+    ///
+    /// assert_eq!(ThinVec::from([1, 2, 3]), thin_vec![1, 2, 3]);
+    /// ```
+    fn from(s: [T; N]) -> ThinVec<T> {
+        std::iter::IntoIterator::into_iter(s).collect()
+    }
+}
+
+impl<T> From<Box<[T]>> for ThinVec<T> {
+    /// Convert a boxed slice into a vector by transferring ownership of
+    /// the existing heap allocation.
+    ///
+    /// **NOTE:** unlike std, this must reallocate to change the layout!
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use thin_vec::{ThinVec, thin_vec};
+    ///
+    /// let b: Box<[i32]> = thin_vec![1, 2, 3].into_iter().collect();
+    /// assert_eq!(ThinVec::from(b), thin_vec![1, 2, 3]);
+    /// ```
+    fn from(s: Box<[T]>) -> Self {
+        // Can just lean on the fact that `Box<[T]>` -> `Vec<T>` is Free.
+        Vec::from(s).into_iter().collect()
+    }
+}
+
+impl<T> From<Vec<T>> for ThinVec<T> {
+    /// Convert a std::Vec into a ThinVec.
```

Comment:
Backticks?

Reply:
More cases below, I won't point them all out, a bit of grepping will find them if you care to change them.